docs/Course Wiki/ECE Course Offerings/ECE470.md

```diff
@@ -0,0 +1,62 @@
+---
+tags:
+- Robotics
+---
+# ECE470
+
+ECE470 (Introduction to Robotics) is a 4-credit-hour course that counts as an Advanced Computing Elective for CEs and as a Hardware Lab for EEs. It is offered in the fall and spring. This course is the same as AE 482 and ME 445.
+
+## Content Covered
+
+- Probablistic state estimation
+- Forward kinematics
+- Inverse kinematics
+- Velocity kinematics
+- Trajectory generation and motion planning
+- Perception and filtering
+
+This course is intended to serve as an introduction to foundations in modern robotics. Concepts such as degrees of freedom, configuration spaces, and constraints are covered along with the differential equations and matrix algebra used to represent them. Then, you will cover frame transformations, forward kinematics (given joint angles, where is the end-effector of the robot?), and velocity kinematics (what joint speeds to we need to make the end-effector move at a certain velocity?). After this, you will cover inverse kinematics (what joint positions do we need to position the end-effector in a specific location?) and briefly discuss trajectory generation, motion planning, and perception and filtering (discrete Bayes filter). In lecture, all of these topics are discussed alongside the differential equations and linear algebra needed to calculate them.
+
+In the lab section, you will mostly focus on implementing the topics from lecture on a UR3 robot arm using ROS and Python. However, for the final lab (as of Spring 2023) you will use OpenCV to detect and stack colored blocks.
+
+There may be additional guest lectures from various robotics-affiliated professors that occur throughout the semester, time permitting. Previous guest lecturers have included:
+
+- Prof. Justin Yim
+- Prof. Holly Golecki
+- Prof. Kris Hauser
+
+## Prerequisites
+
+- One of MATH225, MATH286, MATH415, MATH418
+
+Since MATH286 is no longer offered by the department, it seems that the prerequisites have not been updated. However, this class requires a solid understanding of linear algebra and at least some exposure to differential equations, although there is a review of both at the beginning of the semester. MATH286 has been broken into [MATH257](MATH257.md) (Linear Algebra with Computational Applications) and [MATH285](MATH285.md) (Intro Differential Equations) so taking both of these courses is the recommended prerequisite, although it is possible to take this course without [MATH285](MATH285.md).

```

Comment:
Change to `[MATH257](../MATH%20Course%20Offerings/MATH257.md)` and `[MATH285](../MATH%20Course%20Offerings/MATH285.md)`, otherwise the links won't work.  Also, no need to link math285 the second time.